services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/dataaccess/AccessRequirementManagerImpl.java

```diff
@@ -232,6 +254,7 @@ public AccessRequirement getAccessRequirement(String requirementId) throws Datas
 		for (RestrictableObjectDescriptor rod : toUpdate.getSubjectIds()) {
```

Comment:
We need a delta of the current subjects and the updated subjects to capture the case where a subject ID is removed (the main use case).